### PR TITLE
Make configurable the node builders and dag simplifiers to use

### LINKED ISSUE #68: Allow passing custom simplifiers to the dag builder

### PR DESCRIPTION
**Pull request by eze210**
Closes #68 and #69

### CODE REVIEW

File: spark_board/plan_extractor/transformation_node_builders/repository.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+from .sample_node_builder import SampleNodeBuilder
+from .sort_node_builder import SortNodeBuilder
+from .transformation_node_builder import TransformationNodeBuilder
+from .union_node_builder import UnionNodeBuilder
+from .unknown_node_builder import UnknownNodeBuilder
+from .window_node_builder import WindowNodeBuilder
+
+"""
+Returns the default set of objects that can parse each transformation sub-tree.
+"""
```
Comment (alijdens) [inline spark_board/plan_extractor/transformation_node_builders/repository.py:26]: the doc string goes inside the function (right after the `def`)

File: spark_board/plan_extractor/transformation_node_builders/repository.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    def add_builder(self, node_name: str, builder: TransformationNodeBuilder) -> 'Repository':
+        self.transformation_builders[node_name] = builder
+        return self
+
+    def remove_builder(self, node_name: str) -> 'Repository':
+        self.transformation_builders.pop(node_name)
+        return self
+
+    def with_default_builder(self, provider: Callable[[str], TransformationNodeBuilder]) -> 'Repository':
```
Comment (alijdens) [inline spark_board/plan_extractor/transformation_node_builders/repository.py:84]:
```suggestion
    def with_default_builder(self, provider: Callable[[str], TransformationNodeBuilder]) -> 'Repository':
        """This method sets the callback that will be called whenever a node whose transformation
        type is not known is encountered. This allows defining how those nodes will be handled (which
        information will be shown/extracted from it) or whether to fail, for example."""
```

File: spark_board/plan_extractor/transformation_node_builders/repository.py (hunk 1 above)
Comment (eze210) [reply spark_board/plan_extractor/transformation_node_builders/repository.py]: Oops, java habit...

**Review by alijdens — APPROVED** [review]
LGTM

**Pull request merged**